src/lib/utilities/Drawer/Drawer.svelte

```diff
@@ -48,73 +49,97 @@
 
 	// Local
 	let elemBackdrop: HTMLElement;
+	let windowSettings = { width: 1920, height: 1080 };
+	let styleSettings = { backdrop: '', width: '', height: '' };
+	let animSettings = { x: 0, y: 0 };
 
-	function percentage(percent: number, amount: number): number {
-		return (amount / 100) * percent;
+	// Record a record of default props on init
+	// NOTE: these must stay in sync with the props implemented above.
+	// prettier-ignore
+	const propDefaults = {
+		position, duration,
+		bgBackdrop, blur,
+		bgDrawer, border, rounded, width, height, margin,
+		labelledby, describedby
+	};
+
+	// Listen to drawerStore updates
+	drawerStore.subscribe((settings: DrawerSettings) => {
```

Comment:
You can remove `DrawerSettings` type. The `settings` param should be inferable from there